definitions/checks/repositories/check_upstream_repository.rb

```diff
@@ -0,0 +1,65 @@
+class Checks::CheckUpstreamRepository < ForemanMaintain::Check
+  metadata do
+    label :check_upstream_repository
+    description 'Check if any upstream repositories are enabled on system'
+    tags :pre_upgrade
+    preparation_steps do
+      Procedures::Packages::Install.new(:packages => %w[yum-utils])
+    end
+    confine do
+      feature(:downstream)
+    end
+  end
+
+  def run
+    enabled_upstream_repos = []
+    with_spinner('Checking for presence of upstream repositories') do
+      run_compare(enabled_upstream_repos)
+    end
+    repos_to_disable = repo_label(enabled_upstream_repos)
+    next_steps = []
+    next_steps.concat(
+      [Procedures::Repositories::Disable.new(:repos => repos_to_disable)]
+    )
+    assert(enabled_upstream_repos.empty?,
+           "System has upstream #{repos_to_disable} repositories enabled",
+           :next_steps => next_steps)
```

Comment:
No need to declare extra variable `next_steps` here. 
You can directly assign array `[Procedures::Repositories::Disable.new(:repos => repos_to_disable)]` to `:next_steps` key.